specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/inputs.json

```diff
@@ -0,0 +1,1084 @@
+      "enum": [
+        "LineSeparated",
+        "Array"
+      ],
+      "x-ms-enum": {
+        "name": "JsonOutputSerializationFormat",
+        "modelAsString": true
+      }
+    },
+    "ResourceTestStatus": {
+      "description": "Describes the status of the test operation along with error information, if applicable.",
+      "properties": {
+        "status": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The status of the test operation."
+        },
+        "error": {
+          "readOnly": true,
+          "$ref": "#/definitions/ErrorResponse",
+          "description": "Describes the error that occurred."
+        }
+      }
+    },
+    "ErrorResponse": {
+      "description": "Describes the error that occurred.",
+      "properties": {
+        "code": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Error code associated with the error that occurred."
+        },
+        "message": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Describes the error in detail."
+        }
+      }
+    },
+    "SubResource": {
+      "description": "The base sub-resource model definition.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource Id"
+        },
+        "name": {
+          "type": "string",
+          "description": "Resource name"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "Compression": {
+      "description": "Describes how input data is compressed",
+      "properties": {
+        "type": {
+          "$ref": "#/definitions/CompressionType"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "CompressionType": {
+      "type": "string",
+      "description": "Indicates the type of compression that the input uses. Required on PUT (CreateOrReplace) requests.",
+      "enum": [
+        "None",
+        "GZip",
+        "Deflate"
+      ],
+      "x-ms-enum": {
+        "name": "CompressionType",
+        "modelAsString": true
+      }
+    },
+    "AzureSqlReferenceInputDataSource": {
+      "description": "Describes an Azure SQL database reference input data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/Database",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ReferenceInputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSqlReferenceInputDataSourceProperties"
+        }
+      }
+    },
+    "AzureSqlReferenceInputDataSourceProperties": {
+      "properties": {
+        "server": {
```

Comment:
why doesn't this take a resource ID to the server/database?  How will it know how to connect to a server based on a name?

Reply:
To my knowledge, any user who wants to connect to a sql server only needs to provide these properties (servername, dbname, username, password). We are just using their SDKs: https://docs.microsoft.com/en-us/dotnet/api/overview/azure/sql

Reply:
It may work but its going to require the user to pass around secrets which is generally discouraged.  I won't block on this but I suggest you bring this up in your next thread modeling/security review.  Ideally this uses a managed identity that is attached to your resource to access the SQL server via AAD auth (https://docs.microsoft.com/en-us/azure/azure-sql/database/authentication-aad-overview)